gammapy/analysis/tests/test_analysis.py

```diff
@@ -14,127 +14,86 @@
 
 def test_config():
     config = AnalysisConfig()
-    assert config.settings["general"]["logging"]["level"] == "INFO"
-    cfg = {"general": {"outdir": "test"}}
-    config.update_settings(cfg)
-    assert config.settings["general"]["logging"]["level"] == "INFO"
-    assert config.settings["general"]["outdir"] == "test"
-
-    with pytest.raises(ValueError):
+    assert config.general.log.level == "info"
```

Comment:
Here you just make a config and assert on the default log level.
That doesn't belong in `test_analysis.py`, but in `test_config.py`.
Add the assert on log level default there.

I'm not sure if this `test_config` then is still useful.
Probably you should remove the `config=None` default in `Analysis.__init__` since it's never allowed.
So `config` becomes a required argument.
And then you can just delete this test, or pass some invalid type like `Analysis(config="spam")` and check that a TypeError is raised.

Reply:
There are other tests like `test_docs_file` or `test_config_to_yaml` that only relate to `AnalysisConfig`, `Analysis` isn't needed. Move those over from `test_config.py` to `test_analysis.py`.

Reply:
removed `config=None` in `Analysis.__init__`
and left the test as this:

```
def test_basics():
    config = AnalysisConfig()
    assert "AnalysisConfig" in str(config)
    with pytest.raises(TypeError):
        Analysis("spam")
```

Reply:
moved `test_docs_file` and `test_config_to_yaml` to `test_config.py`